src/app/core/pusher/pusher.service.ts

```diff
@@ -0,0 +1,179 @@
+    });
+
+    this.channel.bind('Project:create', function(data) {
+      thisRef.store.dispatch(
+        new ProjectsAction.CreateSuccess({project: ProjectConverter.fromDto(data.resource, data.parentId)})
+      );
+    });
+    this.channel.bind('Project:update', function(data) {
+      thisRef.store.dispatch(
+        new ProjectsAction.UpdateSuccess({project: ProjectConverter.fromDto(data.resource, data.parentId)})
+      );
+    });
+    this.channel.bind('Project:remove', function(data) {
+      thisRef.store.dispatch(new ProjectsAction.DeleteSuccess({projectId: data.id}));
+    });
+
+    this.channel.bind('View:create', function(data) {
+      thisRef.store.dispatch(new ViewsAction.CreateSuccess({view: ViewConverter.convertToModel(data)}));
+    });
+    this.channel.bind('View:update', function(data) {
+      thisRef.store.dispatch(
+        new ViewsAction.UpdateSuccess({view: ViewConverter.convertToModel(data), skipNotify: true})
+      );
+    });
+    this.channel.bind('View:remove', function(data) {
+      thisRef.store.dispatch(
+        new ViewsAction.DeleteSuccess({viewCode: data.id}) // backend sends code in id in case of View for simplicity
+      );
+    });
+
+    this.channel.bind('Collection:create', function(data) {
+      thisRef.store.dispatch(new CollectionsAction.CreateSuccess({collection: CollectionConverter.fromDto(data)}));
+    });
+    this.channel.bind('Collection:update', function(data) {
+      thisRef.store.dispatch(new CollectionsAction.UpdateSuccess({collection: CollectionConverter.fromDto(data)}));
+    });
+    this.channel.bind('Collection:remove', function(data) {
+      thisRef.store.dispatch(new CollectionsAction.DeleteSuccess({collectionId: data.id}));
+    });
+
+    this.channel.bind('Document:create', function(data) {
+      thisRef.store.dispatch(new DocumentsAction.CreateSuccess({document: convertDocumentDtoToModel(data)}));
+    });
+    this.channel.bind('Document:update', function(data) {
+      thisRef.store.dispatch(new DocumentsAction.UpdateSuccess({document: convertDocumentDtoToModel(data)}));
+    });
+    this.channel.bind('Document:remove', function(data) {
+      thisRef.store.dispatch(new DocumentsAction.DeleteSuccess({documentId: data.id}));
+    });
+
+    this.channel.bind('CompanyContact:update', function(data) {
+      thisRef.store.dispatch(new ContactsAction.GetContactSuccess({contact: ContactConverter.fromDto(data)}));
+    });
+
+    this.channel.bind('ServiceLimits:update', function(data) {
+      thisRef.store.dispatch(
+        new ServiceLimitsAction.GetServiceLimitsSuccess({
+          serviceLimits: ServiceLimitsConverter.fromDto(data.organizationId, data.entity),
+        })
+      );
+    });
+
+    this.channel.bind('Payment:update', function(data) {
+      thisRef.store.dispatch(
+        new PaymentsAction.GetPaymentSuccess({payment: PaymentConverter.fromDto(data.organizationId, data.entity)})
+      );
+    });
+  }
+
+  public ngOnDestroy(): void {
```

Comment:
What is this method good for? AFAIK it is never called by Angular since this is a service and not a component.

The other thing is that this service is a singleton and so unsubscribing is not really needed. Or does server need to know when a user closes the application so it can free some resources?

Reply:
According to available information, this method should be called on services. I wanted a clean unsubscribe. The messages are otherwise sent my way until I fail to respond to ping.